src/main/scala/mesosphere/marathon/core/election/CuratorElectionStream.scala

```diff
@@ -64,23 +61,75 @@ object CuratorElectionStream extends StrictLogging {
       }
   }
 
-  private class CuratorEventEmitter(
-      metrics: Metrics,
-      singleThreadEC: ExecutionContext,
-      clientCloseable: LifeCycledCloseableLike[CuratorFramework],
-      zooKeeperLeaderPath: String,
-      hostPort: String,
-      sq: SourceQueueWithComplete[LeadershipState]) extends Cancellable {
+  private def leaderLatchStream(
+    metrics: Metrics,
+    clientCloseable: LifeCycledCloseableLike[CuratorFramework],
+    zooKeeperLeaderPath: String,
+    hostPort: String,
+    singleThreadEC: ExecutionContext): Source[LeadershipState, Cancellable] = {
+
+    Source.queue[LeadershipState](16, OverflowStrategy.dropHead)
+      .mapMaterializedValue { sq =>
+        val client = clientCloseable.closeable
```

Comment:
Could you please add types to those `val`s?

Reply:
- [ ] Sure